overlord/assertstate/bulk.go

```diff
@@ -35,16 +34,28 @@ import (
 
 const storeGroup = "store assertion"
 
-func bulkRefreshSnapDeclarations(s *state.State, snapStates map[string]*snapstate.SnapState, userID int, deviceCtx snapstate.DeviceContext) error {
-	if len(snapStates) > 512 {
-		return &bulkAssertionFallbackError{errors.New("internal error: more than supported number of snaps")}
-		// TODO: make that work, it's a matter of using many or reusing pools, but keeping only one trail of what is resolved for efficiency
-	}
+var maxGroups = 256
```

Comment:
I think this could use a comment saying that this needs to be a power of 2, and if there is any other reason for the chosen value,

Reply:
that's a fair request, I actually thought about it after I had pushed the PR

Reply:
I discussed with @mvo5, I'll land this and add the comment in the next one